rif-collector-app.php

```diff
@@ -34,4 +34,92 @@ function retrieveAllQuery($db) {
     return $query->fetchAll();
 }
 
-?>
+$message = "";
+$validatedSanitizedArr = [];
+
+function validateSanitizeEntry(array $entryToAdd, &$validatedSanitizedArr, &$message) {
+
+    $dateRegex = '~^\d{4}-\d{2}-\d{2}$~';
+
+    if (!is_string($entryToAdd['make'])) {
+        return $message = "Error with your 'Make' input";
+    }
+    if (!is_string($entryToAdd['model'])) {
+        return $message = "Error with your 'Model' input";
+    }
+    if (!is_string($entryToAdd['type'])) {
+        return $message = "Error with your 'Type' input";
+    }
+    if (!is_string($entryToAdd['color'])) {
+        return $message = "Error with your 'Color' input";
+    }
+    if (!filter_var($entryToAdd['mags'], FILTER_VALIDATE_INT)) {
+        return $message = "Error with your 'Mags' input";
+    }
+    if (!is_string($entryToAdd['power'])) {
+        return $message = "Error with your 'Power' input";
+    }
+    if (!filter_var($entryToAdd['sites'], FILTER_VALIDATE_INT)) {
+        return $message = "Error with your 'Sites Visited' input";
+    }
+    if (!preg_match($dateRegex, $entryToAdd['purchased'])) {
+        return $message = "Error with your 'Purchased Date' input";
+    } else {
+
+        $color = strip_tags($entryToAdd['color']);
+        $color = strtolower($color);
+
+        if ($color === 'white') {
+            $color = 3;
+        } elseif ($entryToAdd['color'] === 'black') {
+            $color = 2;
+        } elseif ($entryToAdd['color'] === 'tan') {
+            $color = 1;
+        } else {
+            return $message = "Error with your 'Color' input";
+        }
+
+    return $validatedSanitizedArr = [
+        'make' => htmlspecialchars($entryToAdd['make']),
+        'model' => htmlspecialchars($entryToAdd['model']),
+        'type' => htmlspecialchars($entryToAdd['type']),
+        'color' => $color,
+        'mags' => filter_var($entryToAdd['mags'], FILTER_SANITIZE_NUMBER_INT),
+        'power' => htmlspecialchars($entryToAdd['power']),
+        'sites' => filter_var($entryToAdd['sites'], FILTER_SANITIZE_NUMBER_INT),
+        'purchased' => ($entryToAdd['purchased']),
```

Comment:
Isn't this kind of double-validating? `filter_var` returns the filtered data if it worked, and false if it didn't. You could hold onto the results of your `filter_vars`. 

`is_string` is technically useless as a validation in this case - everything coming over the wire as `$_POST` will be a string. It would be more useful to run `htmlspecialchars` as your validation, store the output of it, and fail that validation if it ended up as an empty string when the specialchars are removed.